Failure to compile with z3 4.8.5

Using z3 4.8.5, we get the compilation failure:
```
ld: warning: directory not found for option '-L/nix/store/jf934f80x9m0smswv4352l6mvlra1d8n-gcc-wrapper-7.4.0/lib'
Undefined symbols for architecture x86_64:
 "_Z3_get_error_msg_ex", referenced from:
     _c1b9Y_info in libHSalacrity-0.1.0.0-ttwAml4Vgy2aT7WCYfDT6.a(Compiler.o)
     _c1b9J_info in libHSalacrity-0.1.0.0-ttwAml4Vgy2aT7WCYfDT6.a(Compiler.o)
     _c18Gj_info in libHSz3-408.0-1WFnpTfCBrCIgiQLAlveou.a(Base.o)
     _c18Ig_info in libHSz3-408.0-1WFnpTfCBrCIgiQLAlveou.a(Base.o)
     _c18Ng_info in libHSz3-408.0-1WFnpTfCBrCIgiQLAlveou.a(Base.o)
     _c18Q9_info in libHSz3-408.0-1WFnpTfCBrCIgiQLAlveou.a(Base.o)
     _c1ap4_info in libHSz3-408.0-1WFnpTfCBrCIgiQLAlveou.a(Base.o)
     ...                    
ld: symbol(s) not found for architecture x86_64
clang-7: error: linker command failed with exit code 1 (use -v to see invocation)
`cc' failed in phase `Linker'. (Exit code: 1)
```
(The above is the error using clang on a mac, but a similar error happens using gcc on linux.)

--- a/src/Z3/Base.hs
+++ b/src/Z3/Base.hs
@@ -2498,7 +2498,7 @@ z3Error cd = throw . Z3Error cd
 checkError :: Ptr Z3_context -> IO a -> IO a
 checkError cPtr m = do
   m <* (z3_get_error_code cPtr >>= throwZ3Exn)
-  where getErrStr i  = peekCString =<< z3_get_error_msg_ex cPtr i
+  where getErrStr i  = peekCString =<< z3_get_error_msg i
         throwZ3Exn i = when (i /= z3_ok) $ getErrStr i >>= z3Error (toZ3Error i)
 
 ---------------------------------------------------------------------

--- a/src/Z3/Base/C.hsc
+++ b/src/Z3/Base/C.hsc
@@ -1453,10 +1453,6 @@ foreign import ccall unsafe "Z3_set_error"
 foreign import ccall unsafe "Z3_get_error_msg"
     z3_get_error_msg :: Z3_error_code -> IO Z3_string
 
--- | Reference: <http://z3prover.github.io/api/html/group__capi.html#gae0aba52b5738b2ea78e0d6ad67ef1f92>
-foreign import ccall unsafe "Z3_get_error_msg_ex"
-    z3_get_error_msg_ex :: Ptr Z3_context -> Z3_error_code -> IO Z3_string
-
 ---------------------------------------------------------------------
 -- * Miscellaneous
 

